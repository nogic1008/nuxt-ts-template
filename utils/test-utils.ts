/** Generate [0-9a-z]{length} string. */
export const generateRandomString = (length: number) =>
  [...Array(Number.isInteger(length) ? length : 12)]
    .map(() => (~~(Math.random() * 36)).toString(36))
    .join('')
