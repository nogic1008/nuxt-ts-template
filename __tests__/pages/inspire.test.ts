import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import Buefy from 'buefy'
import VueI18n from 'vue-i18n'

import Inspire from '~/pages/inspire.vue'

const localVue = createLocalVue()
localVue.use(Buefy)
localVue.use(VueI18n)

describe('pages/inspire.vue', () => {
  describe.each(['en', 'ja'])('{ locale: %s } snapshot test', (locale) => {
    const i18n = new VueI18n({ locale, silentFallbackWarn: true })
    test('renders correctly', () => {
      const wrapper = mount(Inspire, { localVue, i18n })
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('head()', () => {
    test.each([
      ['en', '"Just start"'],
      ['ja', '"さあ、始めよう"']
    ])('{ locale: %s } returns { title: "%s" }', (locale, title) => {
      // Arrange
      const i18n = new VueI18n({ locale, silentFallbackWarn: true })
      const wrapper = shallowMount(Inspire, { localVue, i18n })

      // Act & Assert
      const head = wrapper.vm.$options.head as Function
      expect(head.call(wrapper.vm)).toStrictEqual({ title })
    })
  })
})
