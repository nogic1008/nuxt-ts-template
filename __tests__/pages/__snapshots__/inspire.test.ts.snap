// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/inspire.vue snapshot renders correctly if locale is "en" 1`] = `
<section
  class="section"
>
  <h2
    class="title is-3 has-text-grey"
  >
    "Just start 
    <span
      class="icon is-large"
    >
      <i
        class="mdi mdi-rocket mdi-48px"
      />
    </span>
    "
  </h2>
   
  <h3
    class="subtitle is-6 has-text-grey"
  >
    Author: 
    <a
      href="https://github.com/anteriovieira"
    >
      Antério Vieira
    </a>
  </h3>
</section>
`;

exports[`pages/inspire.vue snapshot renders correctly if locale is "ja" 1`] = `
<section
  class="section"
>
  <h2
    class="title is-3 has-text-grey"
  >
    "さあ、始めよう 
    <span
      class="icon is-large"
    >
      <i
        class="mdi mdi-rocket mdi-48px"
      />
    </span>
    "
  </h2>
   
  <h3
    class="subtitle is-6 has-text-grey"
  >
    著者: 
    <a
      href="https://github.com/anteriovieira"
    >
      Antério Vieira
    </a>
  </h3>
</section>
`;
